<launch>
  <arg name="node_name" default="gaussian_filter" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show edge image" />

  <arg name="kernel_size" default="11" doc="Defines the size of the kernel to be used" />
  <arg name="sigma_x" default="10" doc="Gaussian kernel standard deviation in X direction." />
  <arg name="sigma_y" default="10" doc="Gaussian kernel standard deviation in Y direction." />

  <!-- gaussian_filter.cpp -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="gaussian_filter" >
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="kernel_size" value="$(arg kernel_size)" />
    <param name="sigma_x" value="$(arg sigma_x)" />
    <param name="sigma_y" value="$(arg sigma_y)" />
  </node>
</launch>
